src/Services/Azure/Kusto/KustoClient.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Text.Json.Nodes;
+using Azure.Core;
+
+namespace AzureMcp.Services.Azure.Kusto;
+
+public class KustoClient
+{
+    #region Private Members
+    private readonly string _clusterUri;
+    private readonly IHttpClientFactory _httpClientFactory;
+    private readonly TokenCredential _tokenCredential;
+    private readonly static string s_application = "AzureMCP";
+    private readonly static string s_clientRequestIdPrefix = "AzMcp";
+    private readonly static string s_default_scope = "https://kusto.dev.kusto.windows.net" + "/.default";
+    #endregion
+
+    #region Public Methods
+    public KustoClient(string clusterUri, IHttpClientFactory httpClientFactory, TokenCredential tokenCredential)
+    {
+        _clusterUri = clusterUri;
+        _httpClientFactory = httpClientFactory;
+        _tokenCredential = tokenCredential;
+    }
+
+    public async Task<KustoResult> ExecuteQueryAsync(string database, string text, CancellationToken cancellationToken)
+    {
+        var uri = _clusterUri + "/v1/rest/query";
+        var httpRequest = await GenerateHttpRequestMessage(uri, database, text, cancellationToken).ConfigureAwait(false);
+
+        using var httpClient = _httpClientFactory.CreateClient();
+        httpClient.BaseAddress = new Uri(_clusterUri);
+        return await SendRequestAsync(httpClient, httpRequest, cancellationToken).ConfigureAwait(false);        
+    }
+
+    public async Task<KustoResult> ExecuteControlCommandAsync(string database, string text, CancellationToken cancellationToken)
+    {
+        var uri = _clusterUri + "/v1/rest/mgmt";
+        var httpRequest = await GenerateHttpRequestMessage(uri, database, text, cancellationToken).ConfigureAwait(false);
+
+        var httpClient = _httpClientFactory.CreateClient();
```

Comment:
we should dispose the client - can we've `using` here as well similar to `ExecuteQueryAsync` method?